src/com/stanimiriliev/chaptertwo/homogeneoustree/Tree.java

```diff
@@ -18,97 +18,141 @@ public Tree(int root) {
 
```

Comment:
Ok and how about the getters and setters. I red that using them inside the class are a good practice, because of the validations, but when I haven't any validations inside them, should I use them or not?